tests/integration/__init__.py

```diff
@@ -1,36 +1,68 @@
 import os
+from contextlib import contextmanager
 
 import pexpect
 
+from code42cli.errors import Code42CLIError
+from code42cli.profile import create_profile
+from code42cli.profile import delete_profile
+from code42cli.profile import get_profile
+from code42cli.profile import switch_default_profile
 
-LINE_FEED = b"\r\n"
-PASSWORD_PROMPT = b"Password: "
-ENCODING_TYPE = "utf-8"
+TEST_PROFILE_NAME = "TEMP-INTEGRATION-TEST"
+_LINE_FEED = b"\r\n"
+_PASSWORD_PROMPT = b"Password: "
+_ENCODING_TYPE = "utf-8"
 
 
-def encode_response(line, encoding_type=ENCODING_TYPE):
-    return line.decode(encoding_type)
+@contextmanager
+def use_temp_profile():
```

Comment:
Can we create this as a fixture? A profile in this application is a product feature and so it makes more sense to have it as a fixture.

Secondly, from design perspective, integration module(directory) is kind of root of the integration tests and it shouldn't  be aware of `code42cli.profile`. i.e. loading it should not invoke application dependencies! At least that is how I see separation of concern. So if not fixture, what if we move this and related methods into a separate file(`util`)?

Reply:
Even if it were in a separate file `util`, it is still (indirectly) invoking the CLI application specific module.

I understand what you are saying, but integration tests already need to have the CLI dependency installed, so I don't see it as a problem for it to important modules there, especially since it is doing it to further advance its purpose in testing it.

Reply:
Yeah agreed, ideally we would create/delete/get/switch profile in an integration test by executing cli commands instead of calling python functions

Reply:
There is also not a real intuitive way to make a context manager a pytest fixture.
The only thing I can think of there is to make the `command_runner` a pytest fixture

Reply:
When testing the profile commands, it makes sense that they are executed via CLI commands... But the intent of this is just to be a context

Reply:
So my idea of separation of concern was in the context that "load modules only when that feature is actually being tested" i.e if we do not want to test the profile feature in certain tests then loading those modules doesn't make sense.

However since profile is used in all features, invoking it here is okay then.

Reply:
We are testing profile commands in this fixture. We are making a clean-slate available for testing the other commands.  The implementation of how that occurs seems irrelevants to me.  It is test setup, not part of the test.

Reply:
If you want a test that tests switching profile, that should be done differently than this.... This is not a means to testing switching profiles.